src/types/Segment.ts

```diff
@@ -91,8 +91,8 @@ export class Segment extends Data implements SegmentProperties {
         this.discontinuity = discontinuity;
         this.mediaSequenceNumber = mediaSequenceNumber;
         this.discontinuitySequence = discontinuitySequence;
-        this.key = key;
-        this.map = map;
+        this.key = key || null;
+        this.map = map || null;
```

Comment:
`?? null`?

Reply:
When hls-parser parses an actual chunklist, the `Segment` object it creates has its `key` and `map` properties set to `null` (unless they're specified, of course). So I'm just trying to better capture the existing behavior